### PR TITLE
Local working

### PR DESCRIPTION
**Pull request by DrBobF**
Some clean up and changed a few items to bring some of the code closer to ANSI standard C.  I ran against some benchmarks here and saw no difference in the results.

### CODE REVIEW

**Comment by wortiz** [comment]
.runguts

File: src/wr_side_data.c [hunk 1]
```diff
@@ -409,7 +413,7 @@ ns_data_print(pp_Data * p,
 
 		/* cos (theta) = n1.n2 / ||n1|| ||n2|| */
 
-		ordinate += 180. - (180./M_PI)*acos((n1[0]*n2[0] + n1[1]*n2[1])/
+		ordinate += 180. - (180./PI)*acos((n1[0]*n2[0] + n1[1]*n2[1])/
```
Comment (wortiz) [inline src/wr_side_data.c]: Would prefer to keep M_PI and use an ifndef M_PI.

Comment (DrBobF) [reply src/wr_side_data.c]: I reverted back to M_PI.

File: src/ac_particles.c [hunk 2]
```diff
@@ -214,6 +214,8 @@ initialize_particles(const Exo_DB * exo,
   int i, j, el_index_begin, el_index_end, num_els, elem_id;
   char time_of_day[80];
   particle_t p;
+  dbl drand48(void);
```
[the remark this answers is not in the document]
Comment (wortiz) [reply src/ac_particles.c]: Are you getting a warning on this from a compiler or something.

My man pages say drand48 is in stdlib.h, maybe you have POSIX features disabled like M_PI?

I'd rather have a `#define _XOPEN_SOURCE` above the `#include <stdlib.h>` in this file if your compiler is giving warnings on this.

http://man7.org/linux/man-pages/man3/drand48.3.html

[the remark this answers is not in the document]
Comment (DrBobF) [reply src/ac_particles.c]: I have gcc 7.3.1 installed. If i read the <stdlib.h> header, it tells me that it is "ISO C99 Standard: 7.20 General utilities". Within, drand48 will require defining _USE_XOPEN or _USE_MISC.

POSIX required inside SNL?

[the remark this answers is not in the document]
Comment (wortiz) [reply src/ac_particles.c]: I don't think there are any strict requirements for POSIX compliance it's just there is at least a few places we use those extensions like the drand48. I don't believe there is an equivalent drand48 without extensions you would have to build it up from rand as far as I remember or use a library.

I'm guessing you were compiling with -std=c99 or -std=c11 and maybe -pedantic on top.

I think the 3 common compilers (gcc, intel, clang) on machines we target all support these extensions so I think the only issue would be if someone were to try to build goma on Windows but that is a whole other headache in itself I'm sure.

**Comment by wortiz** [comment]
.runguts

**Comment by wortiz** [comment]
.runguts

**Pull request merged**